cardano-api/src/Cardano/Api/Value.hs

```diff
@@ -354,9 +345,12 @@ valueFromNestedRep (ValueNestedRep bundles) =
 instance ToJSON ValueNestedRep where
   toJSON (ValueNestedRep bundles) = object $ map toPair bundles
     where
-     toPair :: ValueNestedBundle -> (Text, Aeson.Value)
-     toPair (ValueNestedBundleAda q) = ("lovelace", toJSON q)
-     toPair (ValueNestedBundle pid assets) = (renderPolicyId pid, toJSON assets)
+      toPair :: ValueNestedBundle -> (Text, Aeson.Value)
+      toPair (ValueNestedBundleAda q) = ("lovelace", toJSON q)
+      toPair (ValueNestedBundle pid assets) =
+        ( renderPolicyId pid
+        , toJSON $ Map.mapKeys serialiseToRawBytesHexText assets
```

Comment:
Let's extract this logic to a  `ToJSON AssetName` instance.

Reply:
done